### DISCUSSION TITLE
MATERIAL POINT OUT OF MESH

### DISCUSSION THREAD

**Discussion by MiguelSC98** [Anura3D/Anura3D_OpenSource#106] [Q&A] [answered]
Dear all.
Hi.
I'm modelling large displacement in a natural slope creating two zones and reducing in one the soil strength parameters to model the failure (cohesion from 30 kPa to 0 kPa, angle friction is the same). For that purpose, from two zones, one in the left side is with no variation of soil parameters and oher in right zone is with variation of soil parameters in steps 2 (updating GOM file, activating $$APPLY_MATERIAL_UDATE and then running step 2).
Step 1 Quasi static convergence ON - no variation of soil strength parameters in two zones, the two zones have the same soil strength parameters.
Step 2  Quasi static convergence OFF - variation of strength parameters in right zone.
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/121762035/328cef04-c3bb-451e-a259-ede4c28dba46)

I try to execute from Step 1 (T=0.5 seg) to Step 2- Step 60 (T=0.2 seg) but in step 38, the process failed. (MATERIAL POINT IS OUT FROM THE MESH)
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/121762035/e767d498-b8b1-41cb-a191-bc5582707e3a)
Step 3
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/121762035/2b6e44b0-8b58-419b-85bd-2fe23d1551a7)

Step 6
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/121762035/e4dd0791-ec1f-4100-ba1b-d865b209e17e)

Step 32
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/121762035/e23b2f29-8322-4c47-b2ea-ef13ec98d437)

Step 34
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/121762035/b639cef4-fd7a-4ba4-a2a5-e672779861cf)

Step 35
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/121762035/ea294483-6b09-4468-a36a-9fea76a2a51a)

Step 36
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/121762035/15f3acd0-b50b-480d-a70d-43cb7b518e33)

Step 37
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/121762035/ed4ccedf-2b49-4c8a-ad41-059e4533a250)

Step 38
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/121762035/45d0a079-561c-4937-9e7f-fd67e79de1e7)

According to the developing of surface failure and results in step 38, I have questions about:

> What could be the reason for the material points to come out of the mesh and  how could I solve it?
> What is the reason fto remove  softening Mohr Coulomb model from Anura3D version 2022 since in Anura3D version 2017 it was enabled?

Thank you in advance for your help.

**Comment by EmreDumlu-UM** [comment]
Dear Miguel, 

I can see your mesh is unstructured and very fine. Please be in mind if the proportion of minimum element size to time step (m/s) is less than the maximum velocity of your particle, it may result in instability then so-obtained abnormal points.

Another recommendation I faced before is to try to not apply strain smoothing. 

I assumed you fixed the fixity in the y direction over the domain. I hope it helps,

Best,

**Comment by ayerrocolom** [comment, marked as answer]
Hello,

Your problem of one material point leaving the mesh can be triggered by many different reasons; in particular:
-time step (or Courant number) is too large. You can try to reduce it and see if the simulation becomes more stable.
-existence of empty elements within the continuum triggers numerical instabilities. In your case, the point that flies away comes from a domain where the material points are very separated. I would encourage you to increase the number of points to have a better discretization of the material, specially those zones that you expect some elongation or large deformations.

Regarding your second comment: the aim of Anura3D is to provide an MPM framework. The constitutive models are "external" to the MPM framework, and same models used in FEM can be used in MPM. It is expected that users will use constitutive models compatible with the material they want to model.
